gossip/gossip.go

```diff
@@ -352,6 +352,7 @@ func (g *Gossip) bootstrap() {
 		g.mu.Lock()
 		g.parseBootstrapAddresses()
 		if g.closed {
+			g.mu.Unlock()
```

Comment:
I noticed that there is a more serious-looking bug below in line 372, where the mutex is held while the goroutine sleeps where it looks like it should be the other way around. Could you take care of that as well?